OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -115,7 +115,8 @@ public override bool Tick(Actor self)
 				// TODO: It would be better to not take off at all, but we lack the plumbing to detect current airborne/landed state.
 				// If the aircraft lands when idle and is idle, we let the default idle handler manage this.
 				// TODO: Remove this after fixing all activities to work properly with arbitrary starting altitudes.
-				var skipHeightAdjustment = aircraft.Info.LandWhenIdle && self.CurrentActivity.IsCanceling && self.CurrentActivity.NextActivity == null;
+				var shouldLand = aircraft.Info.IdleBehavior == IdleBehaviorType.Land;
```

Comment:
Imo there is no real benefit to having this as separate var (same below), but not a blocker.

Reply:
I did this mostly due to line length concerns.